kythe/typescript/indexer.ts

```diff
@@ -809,6 +812,32 @@ class Visitor {
     });
   }
 
+  /** forAllInfluencers executes fn for each influencer in the active set. */
+  forAllInfluencers(fn: (influencer: VName) => void) {
+    if (this.influencers.length != 0) {
+      this.influencers[this.influencers.length - 1].forEach(fn);
+    }
+  }
+
+  /** addInfluencer adds influencer to the active set. */
+  addInfluencer(influencer: VName) {
+    if (this.influencers.length != 0) {
```

Comment:
Should it be an error of some sort if we try to add (or forAll()) influencers if nothing is on the stack?

Reply:
I explicitly wanted this to ignore cases where we weren't currently in a context that supports influence (which is currently only inside the body of a `return` statement). Otherwise, the `if (this.influencers.length...` guard would just have to be reproduced everywhere `addInfluencer` is called.